nova/virt/vmwareapi/vm_util.py

```diff
@@ -1524,13 +1524,20 @@ def get_stats_from_cluster(session, cluster):
                                              threads - reserved['vcpus'])
                     max_mem_mb_per_host = max(max_mem_mb_per_host,
                                               mem_mb - reserved['memory_mb'])
+            # NOTE (jakobk): For the total amount of hosts it doesn't matter
+            # whether the host is in MM or unreachable, because the count is
+            # used to calculate safety margins for resource allocations, and MM
+            # or otherwise unreachable hosts is precisely what that is supposed
+            # to guard against.
+            hypervisor_count = len(result.objects)
```

Comment:
I get your point, but I don't like it.

When we expose this metric as `hypervisor_count`, we should be clear that we don't count the number of usable hypervisors, but all hypervisors in the cluster. I would actually expect this to be the number of hypervisors usable by a VM.

I think, it would be better to expose `total_hypervisor_count` and `usable_hypervisor_count` as metrics, to make the difference clear from the beginning, your metric being `total_hypervisor_count`.

We could also get around this, if we don't expose `hypervisor_count` and instead only expose the `stats['mem']['reservable_memory_mb']` by computing it inside this function.

Reply:
Will remove it from the output, and move the calculation here.